model/src/main/kotlin/Vulnerability.kt

```diff
@@ -43,4 +43,37 @@ data class Vulnerability(
      */
     @JsonInclude(JsonInclude.Include.NON_NULL)
     val url: URL? = null
-)
+) {
+    /**
+     * See https://www.first.org/cvss/specification-document#Qualitative-Severity-Rating-Scale.
+     */
+    enum class Rating(private val start: Float, private val endExclusive: Float) {
+        NONE(0.0f, 0.1f),
+        LOW(0.1f, 4.0f),
+        MEDIUM(4.0f, 7.0f),
+        HIGH(7.0f, 9.0f),
+        CRITICAL(9.0f, 10.0f);
+
+        companion object {
+            init {
+                enumValues<Rating>().asIterable().zipWithNext { a, b ->
+                    require(a.start < a.endExclusive) {
+                        "The range [${a.start},${a.endExclusive}[ of $a must be non-empty."
+                    }
+
+                    require(a.endExclusive == b.start) {
+                        "The range of $a needs to end (${a.endExclusive}) where the range of $b starts (${b.start})."
+                    }
+                }
+            }
```

Comment:
The code in init() init will once compiled always have the same effect.
How about moving it into a test class?

Reply:
Done, though slightly differently as I've simplified the constructor parameters.